File: apps/desktop/frontend/src/workspace/jdbcConnectorTypes.ts
export type JdbcDatabaseKind =
  | "vertica"
  | "postgres"
  | "trino"
  | "mysql"
  | "generic_jdbc";

export type JdbcDriverKind = "jdbc" | "generic_jdbc";

export type JdbcConnectorStatus =
  | "not_configured"
  | "configured"
  | "disabled"
  | "error";

export type CreateJdbcConnectorRequest = {
  workspaceId: string;
  displayName: string;
  databaseKind: JdbcDatabaseKind;
  driverKind: JdbcDriverKind;
  jdbcUrlMasked: string;
  environment: string;
  readOnlyDefault?: boolean | null;
  status?: JdbcConnectorStatus | null;
  notes: string;
};

export type ListJdbcConnectorsRequest = {
  workspaceId: string;
};

export type GetJdbcConnectorRequest = {
  workspaceId: string;
  connectorId: string;
};

export type UpdateJdbcConnectorRequest = {
  workspaceId: string;
  connectorId: string;
  displayName: string;
  databaseKind: JdbcDatabaseKind;
  driverKind: JdbcDriverKind;
  jdbcUrlMasked: string;
  environment: string;
  readOnlyDefault: boolean;
  status: JdbcConnectorStatus;
  notes: string;
};

export type JdbcConnector = {
  connectorId: string;
  workspaceId: string;
  displayName: string;
  databaseKind: JdbcDatabaseKind;
  driverKind: JdbcDriverKind;
  jdbcUrlMasked: string;
  environment: string;
  readOnlyDefault: boolean;
  status: JdbcConnectorStatus;
  notes: string;
  createdAt: string;
  updatedAt: string;
  lastUsedAt: string | null;
};

File: crates/hobit-app/src/workspace_service/jdbc_connector_types.rs
#[derive(Clone, Debug, Eq, PartialEq)]
pub struct CreateJdbcConnectorInput {
    pub workspace_id: String,
    pub display_name: String,
    pub database_kind: String,
    pub driver_kind: String,
    pub jdbc_url_masked: String,
    pub environment: String,
    pub read_only_default: Option<bool>,
    pub status: Option<String>,
    pub notes: String,
}

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct UpdateJdbcConnectorInput {
    pub workspace_id: String,
    pub connector_id: String,
    pub display_name: String,
    pub database_kind: String,
    pub driver_kind: String,
    pub jdbc_url_masked: String,
    pub environment: String,
    pub read_only_default: bool,
    pub status: String,
    pub notes: String,
}

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct JdbcConnectorSummary {
    pub connector_id: String,
    pub workspace_id: String,
    pub display_name: String,
    pub database_kind: String,
    pub driver_kind: String,
    pub jdbc_url_masked: String,
    pub environment: String,
    pub read_only_default: bool,
    pub status: String,
    pub notes: String,
    pub created_at: String,
    pub updated_at: String,
    pub last_used_at: Option<String>,
}
